CHANGES.md

```diff
@@ -1,6 +1,417 @@
+* fixed a portability bug in padding years in the DateTime module
+* fixed a bug in dead code elimination relating to local record types
+* fixed a bug comparing floating point expressions on linux32
+* fixed a bug in complicated type aliases
+* fixed a bug in denormalization for '~' for small integers
+* fixed a bug in which remote-value forwarding didn't handle dereferences well
+* fixed a bug relating to scoped accesses to internal modules
+* fixed bugs with parallel iteration over domains with non-natural alignment
+* fixed a bug in the implementation of the &= operator for associative domains
+* fixed a bug in applying `reindex()` to an empty domain/array
+* fixed a bug in modules with just one non-initialization function declaration
+* fixed a bug in dead code elimination for do-while loops
+* fixed a bug in which isRecord*() returned 'true' for sync/single types
+* fixed a bug related to task counters not being stored in task-local storage
+* fixed a bug for ambiguous 'param' methods
+* fixed a bug in 'fifo' tasking in which not enough threads were created
+
+Launchers
+---------
+* added a 'gasnetrun_psm' launcher for running on OmniPath interconnects
+  (see http://chapel.cray.com/docs/master/usingchapel/launcher.html?highlight=launchers#currently-supported-launchers)
+* fixed some bugs in the pbs-gasnetrun_ibv and slurm-based launchers
+
+Generated Code
+--------------
+* cleaned up generated code for no-op statements referencing symbols
+
+Third-Party Software Changes
+----------------------------
+* updated the compiler to be compatible with newer versions of LLVM
+* switched the LLVM back-end to use version 4.0.1 by default
+* updated GASNet to version 1.30.0
+* updated hwloc to version 1.11.8
+* updated GMP to version 6.1.2
+* updated RE2 to commit a810d71
+* enabled suport for multiple communication domains in GASNet for gemini/aries
+
+Testing System
+--------------
+* dramatically improved the quality of the generated web performance graphs
+* improved the redirection of stderr into log files
+
+Developer-oriented changes: Configuration changes
+-------------------------------------------------
+* added support for CHPL_JEMALLOC=system to enable use of a system jemalloc
+* added support for CHPL_HWLOC=system
+
+Developer-oriented changes: Module changes
+------------------------------------------
+* added a "function terminates program" pragma to identify halt() and exit()
+* simplified how automatic 'use's are handled in internal modules
+
+Developer-oriented changes: Makefile improvements
+-------------------------------------------------
+* made our Makefiles respect CFLAGS, CPPFLAGS, and CXXFLAGS
+* we now build the compiler with C++11 when possible for gcc, clang, icc
+
+Developer-oriented changes: Compiler Flags
+------------------------------------------
+* added --user-constructor-error to flag uses of constructors in user code
+* added --force-initializers to compile using initializers over constructors
```

Comment:
I might clarify this to say "to generate default initializers over constructors in some cases", though that might get a bit long.  Right now, it could imply that explicit constructors were impacted, or that we would accept types with both an initializer or a constructor defined by choosing the initializer only, and neither of those interpretations is correct

Reply:
I'm open to suggestions: You've got an 80-column budget and this only needs to make sense for developers.

Reply:
"added --force-initializers to generate default initializers for some types" fits within the character limit, and I think I prefer it.  Thoughts?